edk2toolext/environment/self_describing_environment.py

```diff
@@ -46,8 +52,8 @@ def _gather_env_files(self, ext_strings, base_path):
         # matching the extension.
         matches = {}
         for root, dirs, files in os.walk(base_path, topdown=True):
-            # Check to see whether any of these directories should be skipped.
-            # TODO: Allow these to be passed in via arguments.
+            # Check to see whether any of these directories should be skipped..
+            dirs[:] = [d for d in dirs if pathlib.Path(root, d) not in self.skipped_dirs]
             for index, dir in enumerate(dirs):
```

Comment:
Should the new path check be done inside the existing for loop?  Why do two iterations?